src/main/java/org/elasticsearch/indices/flush/SyncedFlushService.java

```diff
@@ -543,6 +543,9 @@ public ShardId shardId() {
         }
 
         PreSyncedFlushResponse(Engine.CommitId commitId) {
+            if (commitId == null) {
```

Comment:
well we accepted null and further down the road we failed where did expect a non-null value so I added this to fail early